File: n26/format_n26_test.go
package n26

import (
	"os"
	"strings"

	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"
)

var _ = Describe("formatN26", func() {
	Context("In the happy path", func() {
		It("it deletes a few columns from the input", func() {
			sampleN26Csv, _ := os.ReadFile("./fixtures/sample_n26.csv")
			sampleN26Lines := strings.Split(string(sampleN26Csv), "\n")

			formatted, _ := FormatN26(sampleN26Lines)

			expectedResultCsv, _ := os.ReadFile("./fixtures/expected_result.csv")
			expectedResultLines := strings.Split(string(expectedResultCsv), "\n")

			Expect(formatted).To(Equal(expectedResultLines))
		})
	})

	Context("In the sad path", func() {
		It("returns an error if the header string is unexpected", func() {
			badHeaderCsv, _ := os.ReadFile("./fixtures/bad_header_n26.csv")
			badHeaderLines := strings.Split(string(badHeaderCsv), "\n")

      _, err := FormatN26(badHeaderLines)

			Expect(err.Error()).To(Equal("Unexpected header for n26 format"))
		})
	})
})

File: n26/formatN26.test.ts
import { describe, it } from "@std/testing/bdd";
import { expect } from "@std/expect";
import { readCsv } from "../csv/csv.ts";
import { formatN26 } from "./formatN26.ts";

describe("formatN26", () => {
  it("should format n26 csv data correctly", async () => {
    const sampleN26Data = await readCsv("./n26/fixtures/sampleN26.csv", ",", 0);
    const formatted = formatN26(sampleN26Data);
    const expectedResultCsv = await readCsv("./n26/fixtures/expectedResult.csv", ",", 0);
    expect(formatted).toEqual(expectedResultCsv);
  });

  it("should throw error for bad header", async () => {
    const badHeaderData = await readCsv("./n26/fixtures/badHeaderN26.csv", ",", 0);
    expect(() => formatN26(badHeaderData)).toThrow("Unexpected header for n26 format");
  });
});
